### PR TITLE
Support for setting the service version on Log4j2's EcsLayout

### PR DESCRIPTION
**Pull request by tobiasstadler**
## What does this PR do?
Adds support for setting the service version on Log4j2's EcsLayout

## Checklist
- [x] This is an enhancement of existing features, or a new feature in existing plugins
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apm-agent-plugins/apm-ecs-logging-plugin/src/main/java/co/elastic/apm/agent/ecs_logging/Log4j2ServiceVersionInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.declaresMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+
+public class Log4j2ServiceVersionInstrumentation extends TracerAwareInstrumentation {
```
Comment (felixbarny) [inline apm-agent-plugins/apm-ecs-logging-plugin/src/main/java/co/elastic/apm/agent/ecs_logging/Log4j2ServiceVersionInstrumentation.java:40]: Anticipating another instrumentation for service.environment, I think it makes sense to extract an abstract instrumentation that defines `getClassLoaderMatcher` `getTypeMatcher` (which can be extended for example via `return super().and(declaresMethod(named("setServiceVersion")));`), `getMethodMatcher`, and `getInstrumentationGroupNames`.

Comment (tobiasstadler) [reply apm-agent-plugins/apm-ecs-logging-plugin/src/main/java/co/elastic/apm/agent/ecs_logging/Log4j2ServiceVersionInstrumentation.java]: Done

**Review by felixbarny — APPROVED** [review]

**Comment by felixbarny** [comment]
run elasticsearch-ci/docs

**Comment by tobiasstadler** [comment]
Tank You!

**Pull request merged**